book/src/proposals/durable-tx-nonces.md

```diff
@@ -23,30 +23,59 @@ implemented by some CPU ISAs.
 
 ### Contract Mechanics
 
-TODO: svgbob this into a flowchart, add prose
+TODO: svgbob this into a flowchart
 
 ```text
 Start
 Create Account
   state = Uninitialized
-InitializeInstruction(hash)
-  if hash is recent
-    stored_hash = hash
+InitializeInstruction()
```

Comment:
I'm unclear on the need for this method.  The Nonce() instruction can do this work.

Reply:
My thinking was that advancing the state and normal usage ought to be kept separate for complexity's sake.  I suppose they could be combined

Reply:
what is "normal usage" outside advancing the Nonce?

Reply:
Just advancing the nonce and withdraw

Reply:
To elaborate here, combining `Initialize` and `Nonce` will make the result of `Nonce` ambiguous since a "successful" initialization of the program state should not be a "successful" nonce compare-and-swap, as there is nothing to compare.

Reply:
right, but an uninitialized Nonce cannot be verified by the runtime, so the TX will fail unless the Nonce has been initialized

Reply:
That behavior is correct in either case, no?  The questionable one is how to discern between a `Nonce` instruction that successfully initialized an uninitialized account state and a `Nonce` instruction that successfully advanced the nonce value of an initialized account state.  The former should not be aloud to bypass `check_hash_age`, where the latter should.

Reply:
aloud->allowed?

the Nonce() instruction that stores a hash for future use can be packaged in an instruction that uses a normal recent blockhash

Reply:
> aloud->allowed?

📣 

> the Nonce() instruction that stores a hash for future use can be packaged in an instruction that uses a normal recent blockhash

Right.  If the validity checks need to be in runtime anyway, `Nonce` can handle `Initialize`, np.